storage/ksql/src/main/java/io/apicurio/registry/storage/impl/ksql/KafkaSqlRegistryStorage.java

```diff
@@ -0,0 +1,518 @@
+        Runnable runner = () -> {
+            log.info("KSQL consumer thread startup lag: {}", startupLag);
+
+            try {
+                // Startup lag
+                try { Thread.sleep(startupLag); } catch (InterruptedException e) { }
+
+                log.info("Subscribing to {}", topic);
+
+                // Subscribe to the journal topic
+                Collection<String> topics = Collections.singleton(topic);
+                consumer.subscribe(topics);
+
+                // Main consumer loop
+                while (!stopped) {
+                    final ConsumerRecords<String, JournalRecord> records = consumer.poll(Duration.ofMillis(pollTimeout));
+                    if (records != null && !records.isEmpty()) {
+                        log.debug("Consuming {} journal records.", records.count());
+                        records.forEach(record -> {
+                            JournalRecord journalRecord = record.value();
+                            // TODO instead of processing the journal record directly on the consumer thread, instead queue them and have *another* thread process the queue?
+                            processJournalRecord(journalRecord);
+                        });
+                    }
+                }
+            } finally {
+                consumer.close();
+            }
+        };
+        stopped = false;
+        Thread thread = new Thread(runner);
+        thread.setDaemon(true);
+        thread.setName("KSQL Kafka Consumer Thread");
+        thread.start();
+    }
+
+    /**
+     * Process a single journal record found on the Kafka topic.
+     * @param journalRecord
+     */
+    private void processJournalRecord(JournalRecord journalRecord) {
+        log.debug("[{}] Processing journal record of type {}", journalRecord.getUuid(), journalRecord.getMethod());
+        applying.set(Boolean.TRUE);
+        try {
+            Object returnValueOrException = dispatcher.dispatchTo(journalRecord, this);
+            coordinator.notifyResponse(journalRecord.getUuid(), returnValueOrException);
+        } finally {
+            applying.set(Boolean.FALSE);
+        }
+    }
+
+    /**
+     * Creates the Kafka producer.
+     */
+    private KafkaProducer<String, JournalRecord> createKafkaProducer() {
+        Properties props = new Properties();
+
+        // Configure kafka settings
+        props.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        props.putIfAbsent(ProducerConfig.CLIENT_ID_CONFIG, "Producer-" + topic);
+        props.putIfAbsent(ProducerConfig.ACKS_CONFIG, "all");
+        props.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        props.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JournalRecordSerializer.class.getName());
+
+        // Create the Kafka producer
+        KafkaProducer<String, JournalRecord> producer = new KafkaProducer<>(props);
+        return producer;
```

Comment:
Same for producer, use Utils -- it's resilient to exceptions that can be handled ...